checker/src/org/checkerframework/checker/index/qual/IndexOfIndexFor.java

```diff
@@ -0,0 +1,37 @@
+package org.checkerframework.checker.index.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.JavaExpression;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * The annotated expression evaluates to either -1 or a value compatible with a {@link LTLengthOf}
+ * annotation with the same parameters-
+ *
+ * <p>This is the return type of {@link java.lang.String#indexOf(String) String.indexOf} and {@link
+ * java.lang.String#lastIndexOf(String) String.lastIndexOf} in the JDK.
+ *
+ * @checker_framework.manual #index-checker Index Checker
+ */
+@SubtypeOf(IndexOfUnknown.class)
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
+public @interface IndexOfIndexFor {
```

Comment:
The documentation should somewhere explain the name; what is a mnemonic for, or why does this name convey the desired meaning?  I don't understand that from the documentation.